src/tss2-tcti/tcti-device.c

```diff
@@ -144,19 +149,84 @@ tcti_device_receive (
     }
 #endif
     if (response_buffer == NULL) {
+#ifndef TCTI_PARTIAL_READ
         LOG_DEBUG ("Caller queried for size but linux kernel doesn't allow this. "
                    "Returning 4k which is the max size for a response buffer.");
         *response_size = 4096;
         return TSS2_RC_SUCCESS;
     }
+#else
+        /* Read the header only and get the response size out of it */
+        LOG_DEBUG("Partial read - reading response size");
+        fds.fd = tcti_dev->fd;
+        fds.events = POLLIN;
+
+        rc_poll = poll(&fds, nfds, timeout);
+        if (rc_poll < 0) {
+            LOG_ERROR ("Failed to poll for response from fd %d, got errno %d: %s",
+                       tcti_dev->fd, errno, strerror (errno));
+        }
+        if (rc_poll < 0) {
```

Comment:
This looks like duplicate code...

Reply:
Yes, you are right. I'll remove the duplicate.

Reply:
Duplicate code removed.